// @remotion/lambda-client@4.0.332/dist/test/request-handler.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const bun_test_1 = require("bun:test");
(0, bun_test_1.test)('RequestHandler type should be properly exported and defined', () => {
    // Just verify the type can be imported
    const requestHandler = {
        httpsAgent: {
            maxSockets: 100,
        },
    };
    (0, bun_test_1.expect)(requestHandler).toBeDefined();
    (0, bun_test_1.expect)(typeof requestHandler).toBe('object');
});
(0, bun_test_1.test)('API input types should accept requestHandler option', () => {
    // Test that the type checking works for various API inputs
    const getAwsClientInput = {
        region: 'us-east-1',
        service: 'lambda',
        requestHandler: {
            httpsAgent: {
                maxSockets: 50,
            },
        },
    };
    const getFunctionsInput = {
        region: 'us-east-1',
        compatibleOnly: false,
        requestHandler: {
            httpsAgent: {
                maxSockets: 25,
            },
        },
    };
    (0, bun_test_1.expect)(getAwsClientInput.requestHandler).toBeDefined();
    (0, bun_test_1.expect)(getFunctionsInput.requestHandler).toBeDefined();
});
